src/mobile-app/hooks/use-sensor.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect } from "react";
+import { Sensor, ISensorValues, SensorEvent, ISensorConnectionEventData, ISensorValuesEventData, ISensorErrorData } from "../../sensors/sensor";
+import { logError } from "../../shared/utils/log";
+
+export interface IUseSensorResult {
+  connected: boolean;
+  deviceName: string | undefined;
+  values: ISensorValues;
+  error: any | undefined;
+}
+
+export const useSensor = (sensor: Sensor) => {
+
+  const {connected, deviceName, values} = sensor;
+  const [useSensorResult, setUseSensorResult] = useState<IUseSensorResult>({connected, deviceName, values, error: undefined});
```

Comment:
I think its better for code readability to return an object and since we would need to create the object on the return from the hook I think it makes sense to store the internal hook state as an object instead of individual elements..